frontend/src/metabase/static-viz/components/XYChart/utils/ticks.ts

```diff
@@ -151,3 +159,31 @@ export const getYTickWidths = (
         : 0,
   };
 };
+
+export function fixTimeseriesTicksExceedXTickCount(
+  xScaleType: XAxisType,
+  xScale: XScale["scale"],
+  numTicks: number,
+) {
+  const defaultTicks = getTicks(xScale, numTicks);
+
+  if (xScaleType === "timeseries" && defaultTicks.length > numTicks) {
+    let minLengthTicks = defaultTicks;
+    const candidateTickIntervals = [
+      timeWeek.every(2) as TimeInterval,
+      timeMonth.every(2) as TimeInterval,
+    ];
+    candidateTickIntervals
+      .map(tickInterval => getTicks(xScale, tickInterval as unknown as number))
```

Comment:
It's a little funny that visx's `getTicks` expect a number
https://github.com/airbnb/visx/blob/a3b79fd3bae63b100b1a8781f844631a2f3aa2ea/packages/visx-scale/src/utils/getTicks.ts#L5

but as we always land on this return (from my testing)
https://github.com/airbnb/visx/blob/a3b79fd3bae63b100b1a8781f844631a2f3aa2ea/packages/visx-scale/src/utils/getTicks.ts#L14,

we can then pass a `TimeInterval` there instead. Since `d3-time`'s `ticks()` excepts that as well.
https://github.com/d3/d3-time/blob/3e9b59b13e7fc1881365b90797d00d6cb53a0257/src/ticks.js#L44

With all that being said, that's why I need to cast it to `number` to avoid TypeScript complaining about the type here.